server/service/queries.go

```diff
@@ -86,30 +95,55 @@ func listQueriesEndpoint(ctx context.Context, request interface{}, svc fleet.Ser
 	for _, query := range queries {
 		respQueries = append(respQueries, *query)
 	}
+
 	return listQueriesResponse{
 		Queries: respQueries,
+		Count:   count,
+		Meta:    meta,
 	}, nil
 }
 
-func (svc *Service) ListQueries(ctx context.Context, opt fleet.ListOptions, teamID *uint, scheduled *bool, mergeInherited bool) ([]*fleet.Query, error) {
+func (svc *Service) ListQueries(ctx context.Context, opt fleet.ListOptions, teamID *uint, scheduled *bool, mergeInherited bool, urlPlatform *string) ([]*fleet.Query, int, *fleet.PaginationMetadata, error) {
 	// Check the user is allowed to list queries on the given team.
 	if err := svc.authz.Authorize(ctx, &fleet.Query{
 		TeamID: teamID,
 	}, fleet.ActionRead); err != nil {
-		return nil, err
+		return nil, 0, nil, err
 	}
 
-	queries, err := svc.ds.ListQueries(ctx, fleet.ListQueryOptions{
+	// always include metadata for queries
+	opt.IncludeMetadata = true
+
+	var dbPlatform *string
+	if urlPlatform != nil {
+		// validate platform filter
+		if *urlPlatform == "macos" {
+			// More user-friendly API param "macos" is called "darwin" in the datastore
+			dbPlatform = ptr.String("darwin")
+		} else {
+			dbPlatform = urlPlatform
+		}
+		if strings.Contains(*urlPlatform, ",") {
+			return nil, 0, nil, &fleet.BadRequestError{Message: "queries can only be filtered by one platform at a time"}
+		}
+		targetableDBPlatforms := []string{"darwin", "windows", "linux"}
+		if !slices.Contains(targetableDBPlatforms, *dbPlatform) {
+			return nil, 0, nil, &fleet.BadRequestError{Message: "provided platform param cannot be a scheduled query target"}
```

Comment:
Nit, let's provide a little more detail:
```suggestion
			return nil, 0, nil, &fleet.BadRequestError{Message: fmt.Sprintf("platform %q cannot be a scheduled query target, supported platforms are: %s", strings.Join(targetableDBPlatforms, ",")}
```